odl/tomo/data/uncompr_bin.py

```diff
@@ -0,0 +1,472 @@
+        """Read the data from the reader's file and store if desired.
+
+        Parameters
+        ----------
+        dstart : int, optional
+            Offset in bytes of the data field. By default, it is equal
+            to ``header_size``. Negative values are added to the file
+            size in bytes, to support indexing "backwards".
+            Use a value different from ``header_size`` to extract data
+            subsets.
+        dend : `int, optional`
+            End position in bytes until which data is read (exclusive).
+            Negative values are added to the file size in bytes, to support
+            indexing "backwards". Use a value different from the file size
+            to extract data subsets.
+
+        Returns
+        -------
+        data : `numpy.ndarray`
+            The data read from ``self.file``.
+        """
+        filesize_bytes = self.file.seek(0, 2)
+        if dstart is None:
+            dstart_abs = int(self.header_bytes)
+        elif dstart < 0:
+            dstart_abs = filesize_bytes + int(dstart)
+        else:
+            dstart_abs = int(dstart) + self.header_bytes
+
+        if dend is None:
+            dend_abs = int(filesize_bytes)
+        elif dend < 0:
+            dend_abs = int(dend) + filesize_bytes
+        else:
+            dend_abs = int(dend) + self.header_bytes
+
+        if dstart_abs >= dend_abs:
+            raise ValueError('invalid `dstart` and `dend`, resulting in '
+                             'absolute `dstart` >= `dend` ({} >= {})'
+                             ''.format(dstart_abs, dend_abs))
+        if dstart_abs < self.header_bytes:
+            raise ValueError('invalid `dstart`, resulting in absolute '
+                             '`dstart` < `header_bytes` ({} < {})'
+                             ''.format(dstart_abs, self.header_bytes))
+        if dend_abs > filesize_bytes:
+            raise ValueError('invalid `dend`, resulting in absolute '
+                             '`dend` > `filesize_bytes` ({} < {})'
+                             ''.format(dend_abs, filesize_bytes))
+
+        num_elems = (dend_abs - dstart_abs) / self.data_dtype.itemsize
+        if num_elems != int(num_elems):
+            raise ValueError('trying to read {} bytes, corresponding to '
+                             '{} elements of type {}'
+                             ''.format(dend_abs - dstart_abs, num_elems,
+                                       self.data_dtype))
+        self.file.seek(dstart_abs)
+        # TODO: use byteorder according to header
+        data = np.fromfile(self.file, dtype=self.data_dtype,
+                           count=int(num_elems))
+
+        if dstart_abs == self.header_bytes and dend_abs == filesize_bytes:
+            # Full dataset read, reshape to stored shape.
+            # Use 'F' order in reshaping since it's the native MRC data
+            # ordering.
+            data = data.reshape(self.data_shape, order='F')
```

Comment:
A file with this name and module should, i.m.o. have absolutely no references to MRC at all. Easiest if you want to keep a MRC reference (for this pull) is to simply move this class into the MRC module and name it something like `MRC...`.